plugins/register-component/src/components/RegisterComponentPage/RegisterComponentPage.tsx

```diff
@@ -14,50 +14,165 @@
+import { generatePath } from 'react-router';
+
+const useStyles = makeStyles(theme => ({
+  dialogPaper: {
+    minHeight: 250,
+    minWidth: 600,
+  },
+  icon: {
+    width: 20,
+    marginRight: theme.spacing(1),
+  },
+  contentText: {
+    paddingBottom: theme.spacing(2),
+  },
+}));
 
+const FormStates = {
+  Idle: 'idle',
+  Success: 'success',
+  Submitting: 'submitting',
+} as const;
+
+type ValuesOf<T> = T extends Record<any, infer V> ? V : never;
 const RegisterComponentPage: FC<{}> = () => {
-  const [isSubmitting, setIsSubmitting] = useState(false);
+  const classes = useStyles();
+  const catalogApi = useApi(catalogApiRef);
+  const [formState, setFormState] = useState<ValuesOf<typeof FormStates>>(
+    FormStates.Idle,
+  );
 
-  useEffect(() => {
-    if (isSubmitting) {
-      setTimeout(() => {
-        setIsSubmitting(false);
-      }, 4000);
-    }
-  }, [isSubmitting]);
+  const errorApi = useApi(errorApiRef);
+
+  const [result, setResult] = useState<{
+    data: any;
+    error: null | Error;
+  }>({
+    data: null,
+    error: null,
+  });
 
-  const onSubmit = () => {
-    setIsSubmitting(true);
+  const handleSubmit = async (formData: Record<string, string>) => {
+    setFormState(FormStates.Submitting);
+    const { componentLocation: target } = formData;
+    try {
+      const data = await catalogApi.addLocation('github', target);
+
+      setResult({ error: null, data });
+      setFormState(FormStates.Success);
+    } catch (e) {
+      setResult({ error: e, data: null });
+      setFormState(FormStates.Idle);
+      errorApi.post(e);
+    }
   };
 
   return (
     <Page theme={pageTheme.tool}>
+      <Header title="Register existing component" />
       <Content>
-        <ContentHeader title="Register Component">
-          <SupportButton>Documentation</SupportButton>
-        </ContentHeader>
         <Grid container spacing={3} direction="column">
           <Grid item>
             <InfoCard title="Start tracking your component in Backstage">
-              <RegisterComponentForm
-                onSubmit={onSubmit}
-                submitting={isSubmitting}
-              />
+              {formState === FormStates.Submitting ? (
+                <LinearProgress />
+              ) : (
+                <RegisterComponentForm onSubmit={handleSubmit} />
+              )}
             </InfoCard>
           </Grid>
         </Grid>
       </Content>
+      <Dialog
+        open={formState === FormStates.Success}
+        onClose={() => setFormState(FormStates.Idle)}
+        classes={{ paper: classes.dialogPaper }}
+      >
+        <DialogTitle>Component registration result</DialogTitle>
+        {formState === FormStates.Success && (
+          <>
+            <DialogContent>
+              <DialogContentText>
+                Following components have been succefully created:
+                <List>
+                  {result.data.entities.map((entity: any, index: number) => (
```

Comment:
i feel this could have been a data table :)

```
Name               Kind       Type     Namespace
-----------------  ---------  -------  ---------------
my-component-id-1  Component  service  my-namespace
my-component-id-2  Component  website  my-namespace
```

or something like that, and the name is what's clickable to take you to that part of the catalog

Reply:
Could be less visually appealing in case of a single component, which is what we are targeting atm, right?

Reply:
You think it could be? I'm not so sure. Just thinking it'll also be a well known view for the user, and we can use the exact same component as when rendering the other component list type views etc. But I don't have a super strong opinion either way

Reply:
It's just feels that the table is a bit more heavy-weight solution here, with search field, clickable columns etcetc. Let's have it like this and then do a design review session?